crates/katana/node/src/lib.rs

```diff
@@ -151,6 +153,17 @@ impl Node {
         let node_components = (pool, backend, block_producer, validator, self.forked_client.take());
         let rpc = spawn(node_components, self.rpc_config.clone()).await?;
 
+        // --- build and start the gas oracle worker task
+
+        // if the Option<GasWorkerConfig> is none, default to no sampling
+        if let Some(gpo_url) = self.l1_provider_url.as_ref() {
+            let gas_oracle = L1GasOracle::sampled(Some(gpo_url.clone())); // Clone the URL if needed
```

Comment:
we dont need to match on the `l1_provider_url` here, considering we already created the `L1GasOracle` earlier. so simply run the worker here.